component.json

```diff
@@ -380,6 +381,16 @@
       "metadata": {
         "in": {}
       }
+    },
+    "upsertFile": {
+      "main": "./lib/actions/upsertFile.js",
+      "title": "Write File to S3 From a Provided Attachment",
+      "help": {
+        "description": "Given a filename and a URL to an attachment stored in the platform, transfers the contents of the attachment to AWS S3. The component returns a summary of the written file. AWS S3 always overwrites the contents of the file if it already exists.",
+        "link": "/components/aws-s3/index.html#write-file-to-s3-from-a-provided-attachment"
+      },
+      "fields": {},
```

Comment:
No fields at all? What about bucket?

Reply:
Bucket exists as a field on the incoming message/metadata.  This way, if an integrator needs to dynamically route files to different buckets, they can do so. If they are only writing to a single bucket, they can select a valid enum member.